lib/linter/linter.js

```diff
@@ -469,16 +470,15 @@ const eslintEnvPatterns = [
 function findEslintEnv(text) {
     let match, retv;
 
-    for (const eslintEnvPattern of eslintEnvPatterns) {
-        eslintEnvPattern.lastIndex = 0;
+    eslintEnvPattern.lastIndex = 0;
 
-        while ((match = eslintEnvPattern.exec(text)) !== null) {
-            retv = Object.assign(
-                retv || {},
-                commentParser.parseListConfig(stripDirectiveComment(match[1]))
-            );
-        }
+    while ((match = eslintEnvPattern.exec(text)) !== null) {
+        retv = Object.assign(
+            retv || {},
+            commentParser.parseListConfig(stripDirectiveComment(match[1]))
+        );
```

Comment:
```suggestion
        if (match[0].endsWith("*/")) {
            retv = Object.assign(
                retv || {},
                commentParser.parseListConfig(stripDirectiveComment(match[1]))
            );
        }
```

Just for better reasoning about the new regex. I don't think we need to add a test for this, as it would be a parsing error.